orbit/cmd/desktop/desktop.go

```diff
@@ -119,23 +137,53 @@ func main() {
 			}()
 
 			return done
-		}()
+		}
+
+		// start a check as soon as the app starts
+		deviceEnabledChan := checkToken()
 
+		// this loop checks the `mtime` value of the token file and:
+		// 1. if the token file was modified, it disables the tray items until we
+		// verify the token is valid
+		// 2. calls (blocking) `checkToken` to verify the token is valid
 		go func() {
 			<-deviceEnabledChan
-			tic := time.NewTicker(5 * time.Minute)
+			tic := time.NewTicker(1 * time.Second)
 			defer tic.Stop()
 
 			for {
 				<-tic.C
+				expired, err := tokenReader.HasChanged()
+				switch {
+				case err != nil:
+					log.Error().Err(err).Msg("check token file")
+				case expired:
+					log.Info().Msg("token file changed, rechecking")
+					disableTray()
+					<-checkToken()
+				}
+			}
+		}()
+
+		// poll the server to check the policy status of the host and update the
```

Comment:
<No need to tackle this now :)>

As an optimization, if product team is a-ok with polling for "policy status" every `1h` instead of every `5m`, then we can re-use the `checkToken` we do on every token rotation (the above routine) to update the policy status. (We feed two birds with one scone.)

Reply:
this sounds like a great idea, 1h shouldn't be too much of a delay in my opinion, but agreed with what you said: let's check and update separately